### PR TITLE
Func snippet

### PR DESCRIPTION
**Pull request by AlexandrLitkevich**

### CODE REVIEW

File: src/preview/index.js [hunk 1]
```diff
@@ -27,17 +27,20 @@ const withData = (ThemeProvider, { providerFn, onThemeSwitch }) => {
...
     onThemeSwitch: () => onThemeSwitch,
+
   })(DecoratorUI(CurrentThemeProvider), { isGlobal: true });
 };
 
 export const withThemes = (
   ThemeProvider,
   themesList,
   { providerFn, onThemeSwitch = onThemeSwitchDefault } = {},
+  snippetFunc,
```
Comment (usulpro) [inline src/preview/index.js:38]: This is good. But I'd like to align this with a future planning features and with the current API style. So let's update it in the following way:

1. Move it into the third agrument, so it's more convinient to keep optional configs in a single place
2. Let's sligtly update a name. We plan to have the similar feature for the values panel, so I'd like it to be a bit more specific e.g. `fieldSnippetFn`
3. We can pass a default value right here and rely on this in other places

File: .storybook/stories.js [hunk 2]
```diff
@@ -25,12 +25,19 @@ export const onThemeSwitch = context => {
   };
 };
 
+const funcSnippet = (selectedValue) => {
+  console.log("funcSnippet", selectedValue)
```
Comment (usulpro) [inline .storybook/stories.js:29]: if this is temporary code only for debugging let's remove it before megring

File: src/manager/components/ThemeBrowser.js [hunk 3]
```diff
@@ -1,32 +1,32 @@
...
+const showThemePath = (selectedValue, snippetFunc) => {
   if (!selectedValue) return 'select value';
   try {
-    const { namespace, name } = selectedValue;
-    const path = namespace.join('.');
-    const fullPath = `${path}.${name}`;
-    const themeProp = `\${({ theme }) => theme.${fullPath}}`;
-    return themeProp;
+    if (snippetFunc) return snippetFunc(selectedValue);
+    return defaultSnippet(selectedValue);
```
Comment (usulpro) [inline src/manager/components/ThemeBrowser.js:19]: See my comment in the `preview/index.js` - if we pass the default value - wouldn't need to check if it exists

File: src/manager/components/ThemeBrowser.js [hunk 4]
```diff
@@ -1,32 +1,32 @@
 import React from 'react';
 import ReactJson from '@usulpro/react-json-view';
-import { useTheme } from '@storybook/theming';
+import {useTheme} from '@storybook/theming';
```
Comment (usulpro) [inline src/manager/components/ThemeBrowser.js:3]: I see the changes like this. It looks like you IDE didn't catch project settings. We should try to keep the existing code style. If your IDE can't solve this - the easiest way is to launch `yarn lint:fix` in the terminal to restore styles

File: README.md [hunk 5]
```diff
@@ -197,6 +197,43 @@ ReactDOM.render(
 There is an example app with CRA, Material-UI and Storybook Addon [Demo](https://react-theming.github.io/theming-material-ui/) [Source](https://github.com/react-theming/theming-material-ui)
 
 
+## Use your output of the selected value
```
Comment (usulpro) [inline README.md:200]: Thanks for adding documentation for the new feature. Let's put it to the section where all features are described - I think next to the "background colors" would be good

**Review by usulpro — CHANGES_REQUESTED** [review]
Good first PR! Thanks a lot. Just fix some minor issues - see it in my comments

File: src/preview/index.js (hunk 1 above)
Comment (AlexandrLitkevich) [reply src/preview/index.js:38]: Done

File: .storybook/stories.js (hunk 2 above)
Comment (AlexandrLitkevich) [reply .storybook/stories.js:29]: Done

File: src/manager/components/ThemeBrowser.js (hunk 3 above)
Comment (AlexandrLitkevich) [reply src/manager/components/ThemeBrowser.js:19]: Done

File: src/manager/components/ThemeBrowser.js (hunk 4 above)
Comment (AlexandrLitkevich) [reply src/manager/components/ThemeBrowser.js:3]: Done

File: README.md (hunk 5 above)
Comment (AlexandrLitkevich) [reply README.md:200]: Done

File: src/preview/index.js (hunk 1 above)
Comment (usulpro) [reply src/preview/index.js]: it should be

```
{
      providerFn,
      onThemeSwitch = onThemeSwitchDefault,
     fieldSnippetFn  = fieldSnippetFnDefault,
}
```

**Comment by usulpro** [comment]
Great! Thanks @AlexandrLitkevich

**Pull request merged**